src/operators_lazyproduct.jl

```diff
@@ -62,28 +73,21 @@ permutesystems(op::LazyProduct, perm::Vector{Int}) = LazyProduct(([permutesystem
 identityoperator(::Type{LazyProduct}, ::Type{S}, b1::Basis, b2::Basis) where S<:Number = LazyProduct(identityoperator(S, b1, b2))
 
 
-# Fast in-place multiplication
 function mul!(result::Ket{B1},a::LazyProduct{B1,B2},b::Ket{B2},alpha,beta) where {B1,B2}
-    tmp1 = Ket(a.operators[end].basis_l)
-    mul!(tmp1,a.operators[end],b,a.factor,0)
+    mul!(a.ket_l[end],a.operators[end],b,a.factor,0)
     for i=length(a.operators)-1:-1:2
-        tmp2 = Ket(a.operators[i].basis_l)
-        mul!(tmp2,a.operators[i],tmp1)
-        tmp1 = tmp2
+        mul!(a.ket_l[i],a.operators[i],a.ket_l[i+1])
     end
-    mul!(result,a.operators[1],tmp1,alpha,beta)
+    mul!(result,a.operators[1],a.ket_l[2],alpha,beta)
```

Comment:
You're actually never using `ket_l[1]` (as it directly goes into the result). Can you remove it, i.e. never store it at all in the construction?

Reply:
There is also a test failure here, which happens when `length(ket_l) == 1`, so indexing at `2` errors.

Reply:
Yes, it seems cause index wrong. I am trying to fix it. And I notice that in the master branch. For one operator `LazyProduct`, it seems `mul!` twice.